File: prueba.py/convertor.js
// Convertor de Unidades en JavaScript

function mostrarMenu() {
    console.log("");
    console.log("---------------------------------------------------- Bienvenid@ al convertor de unidades ----------------------------------------------------");
    console.log("");
    console.log("para CELSIUS -> FAHRENHEIT coloque 1");
    console.log("para METROS -> KILOMETROS coloque 2");
    console.log("");
}

// Función para pedir un número válido
function pedirNumero(mensaje) {
    while (true) {
        let entrada = prompt(mensaje); // prompt devuelve string
        let numero = parseFloat(entrada); // convertir a número

        if (!isNaN(numero)) {
            return numero; // si es válido, lo devuelve
        } else {
            console.log("Error: debe ingresar un número válido. Intente de nuevo.");
        }
    }
}

// Programa principal
mostrarMenu();
let opcion = pedirNumero("Seleccione una opción (1 o 2):");

if (opcion === 1) {
    let celsius = pedirNumero("Ponga la temperatura en Celsius:");
    let resultado = (9/5 * celsius) + 32;
    console.log(`${celsius} °C en Fahrenheit son: ${resultado} °F`);
} else if (opcion === 2) {
    let metros = pedirNumero("Ponga los metros a convertir:");
    let resultado = metros / 1000;
    console.log(`${metros} mts en kilómetros son: ${resultado} km`);
} else {
    console.log("Formato incorrecto, solo se toman en cuenta las opciones 1 y 2");
}

console.log("");
console.log("Fin del programa, ¡ten un feliz resto de día! :)");

File: prueba.py/convertor.py
bandera = True
while bandera:
    print("")
    print("---------------------------------------------------- Bienvenid@ al convertor de unidades ----------------------------------------------------")
    print("")
    print("para CELSIUS -> FAHRENHEIT coloque 1")
    print("para METROS -> KILOMETROS coloque 2")
    print("")

    while True:
        try:
            OPCION = int(input("Seleccione una opción: "))
            break
        except ValueError:
            print("Error: debe ingresar un número (1 o 2) Intente de nuevo.")

    if OPCION == 1:
        while True:
            try:
                Celsius = float(input("Ponga la temperatura en Celsius: "))
                resultado = (9/5 * Celsius) + 32
                print("")
                print(Celsius, "°C en Fahrenheit son:", resultado, "°F")
                break
            except ValueError:
                print("Error: solo se permiten números. Intente de nuevo")
    elif OPCION == 2:
        while True:
            try:
                metros = float(input("Ponga los metros a convertir: "))
                resultado = metros / 1000
                print("")
                print(metros, "mts en kilómetros son:", resultado, "km")
                break
            except ValueError:
                print("Error: solo se permiten números. Intente de nuevo")
    else:
        print("Formato incorrecto, solo se toman en cuenta las opciones 1 y 2")

    bandera = False

print("")
print("Fin del programa, ¡ten un feliz resto de día! :)")